File: mode.js
const input = require("./input");

(async () => {
    const numbers = await input("Enter numbers (Separated by Comma): ");
    const num_list = numbers.split(",").map(Number);

    // Count occurrences
    let instancesCount = {};
    num_list.forEach(num => {
        if (instancesCount[num]) {
            instancesCount[num] += 1;
        } else {
            instancesCount[num] = 1;
        }
    });

    // Display counts
    console.log("Count of all numbers:");
    for (let [num, count] of Object.entries(instancesCount)) {
        console.log(`${num}: ${count}`);
    }

    // Find the max count
    let maxCount = 0;
    for (let num in instancesCount) {
        if (instancesCount[num] > maxCount) {
            maxCount = instancesCount[num];
        }
    }

    // Collect all modes
    let modes = [];
    for (let num in instancesCount) {
        if (instancesCount[num] === maxCount) {
            modes[modes.length] = num; // push manually
        }
    }

    console.log(`Mode(s): ${modes.join(", ")}`);
})();

File: mode.py
# Get input from the user
numbers = input("Enter numbers (Separated by Comma): ")
num_list = [int(num) for num in numbers.split(",")]

# Count occurrences
instances_count = {}
for num in num_list:
    if num in instances_count:
        instances_count[num] += 1
    else:
        instances_count[num] = 1

# Display counts
print("Count of all numbers:")
for num, count in instances_count.items():
    print(f"{num}: {count}")

# Find the max count manually
max_count = 0
for count in instances_count.values():
    if count > max_count:
        max_count = count

# Collect all modes
modes = []
for num, count in instances_count.items():
    if count == max_count:
        modes.append(str(num))  # convert to string for printing

print(f"Mode(s): {', '.join(modes)}")